Implement light/pick_divisible_coins_for_amount method

bonustrack opened the pull request:
#### Changes:

Added a new WS API method called `light/pick_divisible_coins_for_amount` and routed with the function `composer.pickDivisibleCoinsForAmount` as suggested on #103.

Fix #103

tonyofbyteball commented:
right direction but a few changes are necessary:
1) please don't introduce cyclic dependencies, composer also require()s network.  Maybe pickDivisibleCoinsForAmount should be moved to a separate module.
2) to make it easier for the client, pass asset rather than object, read the obj by asset
3) also validate types of params, otherwise the light vendor could be crashed by a malicious request.

bonustrack commented:
Thank you for you review @tonyofbyteball. So i've done these changes:
1. Moved `pickDivisibleCoinsForAmount` from composer.js to light.js. About this, would you like to keep the method accessible in composer.js with a route to light.js?
2. I've changed `obj_asset` object to `asset` string. I see that the method `pickDivisibleCoinsForAmount` allow passing asset params like `is_private` `cap` or `issued_by_definer_only` https://github.com/byteball/byteballcore/blob/5e1ffe750ca784ca992c45417c9ea84f2873ff48/composer.js#L71 i'm not sure if we want to handle these params.
3. Now it also validate every params type.

Please let me know if you want me to do some changes.

tonyofbyteball commented:
1. This function is common for all wallets, not just light ones.  I think it deserves its own module, call it e.g. inputs.js.

2. Sorry I was not clear, I suggested using storage.readAssetInfo, then you have all required asset properties.

3. That's much better but last_ball_mci and amount must be also positive integers, use isPositiveInteger from validation_utils.js.

bonustrack commented:
Thank you for the details, i think i get it now, you can review the changes again @tonyofbyteball
